runtime/common/src/account_conversion.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_primitives::AccountId;
+use cfg_types::domain_address::DomainAddress;
+use pallet_evm::AddressMapping;
+use sp_core::{Get, H160};
+use sp_runtime::traits::Convert;
+
+/// Common converter code for translating accounts across different
+/// domains and chains.
+pub struct AccountConverter<R>(core::marker::PhantomData<R>);
+
+impl<R> AccountConverter<R> {
+	/// Converts an EVM address from a given chain into a local AccountId
+	fn convert_evm_address(chain_id: u64, address: [u8; 20]) -> AccountId {
+		let tag = b"EVM\0";
+		let mut bytes = [0; 32];
+		bytes[0..20].copy_from_slice(&address);
+		bytes[20..28].copy_from_slice(&chain_id.to_be_bytes());
+		bytes[28..32].copy_from_slice(tag);
+		AccountId::new(bytes)
+	}
+}
+
+// Implement EVM account conversion using our shared conversion code
+impl<R> AddressMapping<AccountId> for AccountConverter<R>
+where
+	R: pallet_evm_chain_id::Config,
+{
+	fn into_account_id(address: H160) -> AccountId {
+		let chain_id = pallet_evm_chain_id::Pallet::<R>::get();
+		Self::convert_evm_address(chain_id, address.0)
+	}
+}
+
+// Implement connectors account conversion using our shared conversion code
+impl<R> Convert<DomainAddress, AccountId> for AccountConverter<R> {
+	fn convert(domain_address: DomainAddress) -> AccountId {
+		match domain_address {
+			DomainAddress::Centrifuge(addr) => AccountId::new(addr),
```

Comment:
@NunoAlexandre this behavior is different from what `DomainAddress::into_address_truncating` did. This is a direct unwrapping, whereas `into_address_truncating` would have done something like `[<tag_bytes>,<truncated_address>]`. I didn't see anything in the comments or commit messages that made that behavior seem intentional, and there were no tests that covered it.

We can change this so that `DomainAddress::Centrifuge` encodes into something different if we so desire

Reply:
Ah, good point. For all I know, this is exactly what we want since the idea of `DomainAddress::Centrifuge(address)` is exactly to represent a direct account address on the Centrifuge chain.